[Filebeat][GSuite] Initial implementation of SAML and User Accounts filesets (#19329)
elastic/beats: x-pack/filebeat/module/gsuite/_meta/docs.asciidoc

[role="xpack"]

:modulename: gsuite
:has-dashboards: true

== GSuite module

beta[]

This is a module for ingesting data from the different GSuite audit reports API's.

include::../include/gs-link.asciidoc[]

[float]
=== Compatibility

It is compatible with a subset of applications under the https://developers.google.com/admin-sdk/reports/v1/get-start/getting-started[Google Reports API v1]. As of today it supports:

- https://developers.google.com/admin-sdk/reports/v1/appendix/activity/saml[SAML Audit Activity Events]
- https://developers.google.com/admin-sdk/reports/v1/appendix/activity/user-accounts[User Accounts Activity Events]

=== Configure the module

> To allow the filebeat module to ingest data from the Google Reports API, you would need an admin account, and set up a `ServiceAccount` to access `Admin SDK API`. Also notice, https://developers.google.com/admin-sdk/reports/v1/guides/delegation[Domain-Wide Delegation] is required for your application in order to work properly.

Reviewer: The first sentence of this reads a little strange. How about?

```suggestion
In order for filebeat to ingest data from the Google Reports API you must set up a `ServiceAccount` that has access to the `Admin SDK API`. Additionally https://developers.google.com/admin-sdk/reports/v1/guides/delegation[Domain-Wide Delegation] is required for your application to work properly.
```

This is a list of GSuite Reports fields that are mapped to ECS.

[options="header"]
|=======================================================================================
| GSuite Reports             | ECS Fields                                              |
| items[].id.time            | @timestamp                                              |
| items[].id.uniqueQualifier | event.id                                                |
| items[].id.applicationName | event.provider                                          |
| items[].events[].name      | event.action                                            |
| items[].customerId         | organization.id                                         |
| items[].ipAddress          | client.ip, related.ip, client.as.*, client.geo.*        |
| items[].actor.email        | client.user.email, client.user.name, client.user.domain |
| items[].actor.profileId    | client.user.id                                          |
|=======================================================================================

These are the common ones to all filesets.

[float]
=== Example dashboard

This module comes with a sample dashboard. For example:

> TODO: include an image of a sample dashboard. If you do not include a dashboard,

Reviewer: Is this TODO still valid or does this and the surrounding few lines need to be removed?

Author: Plan to add a dashboard later, but will remove the section in the meanwhile :+1:
